spec/shared/controllers/hiding_columns_tenant.rb

```diff
@@ -0,0 +1,39 @@
+shared_examples 'hiding tenant columns' do
+
+  before do
+    allow(controller).to receive(:render)
+    login_as user
+  end
+
+  subject do
+    expect(controller).to receive(:get_db_view).and_return(report)
+    controller.send(:report_data)
+    controller.send(:view_to_hash, assigns(:view))
+  end
+
+  context "when not admin" do
+    let(:user) { FactoryBot.create(:user) }
+    it "renders show_list and does not include hidden column(hidden by display method)" do
+      headers = subject[:head].map { |x| x[:text] }.compact
+      expect(headers).to match_array(result_header)
+      expect(headers).not_to include('Tenant')
+
+      first_row_values = subject[:rows][0][:cells].map { |x| x[:text] }.compact
+      expect(first_row_values).to match_array(result_array)
+      expect(first_row_values).not_to include(not_in_array)
```

Comment:
You don't really need this. The `match_array` makes sure that nothing else is included in the array.